src/org/opensolaris/opengrok/analysis/sh/ShSymbolTokenizer.lex

```diff
@@ -38,48 +37,65 @@ super(in);
 %include CommonTokenizer.lexh
 %char
 
-Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
-
 %state STRING COMMENT SCOMMENT QSTRING
 
+%include Common.lexh
+%include Sh.lexh
 %%
 
 <YYINITIAL> {
-{Identifier} {String id = yytext();
+{Identifier}    {
+    String id = yytext();
                 if(!Consts.shkwd.contains(id)){
                         setAttribs(id, yychar, yychar + yylength());
                         return yystate(); }
               }
+ {Number}    {}
  \"     { yybegin(STRING); }
  \'     { yybegin(QSTRING); }
  "#"    { yybegin(SCOMMENT); }
+
+ {Unary_op_req_lookahead} / \W    {
```

Comment:
if it's a noop why is it here?

Reply:
I think @kahatlen experimented a bit with some lookaheads in jflex, but not sure how good they worked ...

Reply:
The handler is a noop, but the match is not. E.g. this match consumes characters so they aren't later processed by the {Identifier} rule; I wanted to prevent unary ops from producing symbols.

I haven't seen any problems with lookaheads. They were already used in other lexers — e.g. here in ShXref `"${" / {WhspChar} | {EOL}`, and used in Scala, Perl, Ada, Kotlin and Swift.

Reply:
It aligns with the same (new) rule in ShXref.

Reply:
why would they produce symbols?
(when they don't match the identifier rule ? )
can you give an example of what text you're trying to get avoided to be matched as identifier?

Reply:
ok, I typed the Q too fast ... I see now
so e.g.
-a -f -x 
but ... what about positive case?
isn't there any?

Reply:
I referenced the Bash getopts documentation, but you're right that ksh supports '+' as an option prefix. I can update Unary_op_req_lookahead to support that syntax too.